tfx/components/example_gen/utils.py

```diff
@@ -218,42 +249,49 @@ def _retrieve_latest_span(uri: Text,
     raise ValueError('Only one {SPAN} is allowed in %s' % split_pattern)
 
   split_glob_pattern = split_pattern.replace(SPAN_SPEC, '*')
+  split_glob_pattern = split_glob_pattern.replace(VERSION_SPEC, '*')
   logging.info('Glob pattern for split %s: %s', split.name, split_glob_pattern)
   split_regex_pattern = _glob_to_regex(split_pattern).replace(SPAN_SPEC, '(.*)')
+  split_regex_pattern = split_regex_pattern.replace(VERSION_SPEC, '.*')
   logging.info('Regex pattern for split %s: %s', split.name,
                split_regex_pattern)
-  if re.compile(split_regex_pattern).groups != 1:
-    raise ValueError('Regex should have only one group')
 
-  files = tf.io.gfile.glob(split_glob_pattern)
-  latest_span = None
-  for file_path in files:
-    result = re.search(split_regex_pattern, file_path)
-    if result is None:
-      raise ValueError('Glob pattern does not match regex pattern')
-    try:
-      span = int(result.group(1))
-    except ValueError:
-      raise ValueError('Cannot not find span number from %s based on %s' %
-                       (file_path, split_regex_pattern))
-    if latest_span is None or span >= int(latest_span):
-      # Uses str instead of int because of zero padding digits.
-      latest_span = result.group(1)
+  return _property_search(split, split_glob_pattern, split_regex_pattern,
+                          SPAN_PROPERTY_NAME, SPAN_SPEC)
 
-  if latest_span is None:
-    raise ValueError('Cannot not find matching for split %s based on %s' %
-                     (split.name, split.pattern))
-  return latest_span
 
+def _retrieve_latest_version_from_span(uri: Text,
```

Comment:
can we merge _retrieve_latest_version_from_span  & _retrieve_latest_span to reduce duplication?